nextflow.config

```diff
@@ -114,6 +112,16 @@ params {
   config_profile_contact = false
   config_profile_url = false
 
+  // HBA-DEALS
+  skip_hbadeals = false
+  rsem_file_suffix  = '.isoforms.results'
+  hbadeals_isoform_level = 'TRUE'
+  hbadeals_mcmc_iter = 1000
+  hbadeals_mcmc_warmup = 200
```

Comment:
**NOTE:** 
Recommended values for our real run by @karleg :

```R
  hbadeals_mcmc_warmup = 200000
  hbadeals_mcmc_iter = 1000000

```

Reply:
The values I used for the runs in this projects are warmup=20000 and iter=1000000, so I think there is one 0 too much in the warmup here

Reply:
Noted @karleg ! Thanks for quick feedback, I am running the last tests before deploying the final run.


Here's one job with 
N = 10 (6 cases, 4 controls) from this file, cohort 1 [case_control-april21.tsv](https://github.com/TheJacksonLaboratory/coronavirus-sample-pre-processing-nf/blob/655ccc8aead83ecd40168fa83825f2f87c3cf28a/data/case_control-april21.tsv).

I use very low iters to check inputs for the test.

Everything seems to be running well, apart from some warnings. Not sure how alarming these are or if it's usual. Do you mind taking a look?

```console
Warning messages:
1: In regularize.values(x, y, ties, missing(ties), na.rm = na.rm) :
  collapsing to unique 'x' values
2: In regularize.values(x, y, ties, missing(ties), na.rm = na.rm) :
  collapsing to unique 'x' values
3: In mclapply(1:length(unique(countsData[, 1])), function(gene.number) { :
  scheduled cores 1, 5, 6, 8, 12, 28 did not deliver results, all values of the jobs will be affected
```

The data used for calling `hbadeals::hbadeals()` (`labels`, `countsData`) are in the folder `results` > `hbadeals` > `contrast_cohort_1__Nctrl_6__Ncases_4`  here: 

🔗 https://cloudos.lifebit.ai/public/jobs/5eb335791795470103dd1e4e

![image](https://user-images.githubusercontent.com/38183826/81238453-0b282e00-8ffa-11ea-9774-d0d62cbf041e.png)

I am writing the dataframes right before calling `hbadeals::hbadeals()`

https://github.com/TheJacksonLaboratory/coronavirus-nf/blob/877f96aa10cad5ee84af52439a83b8f8deedac7b/containers/hbadeals/rsem2hbadeals.R#L218

https://github.com/TheJacksonLaboratory/coronavirus-nf/blob/877f96aa10cad5ee84af52439a83b8f8deedac7b/containers/hbadeals/rsem2hbadeals.R#L225

https://github.com/TheJacksonLaboratory/coronavirus-nf/blob/877f96aa10cad5ee84af52439a83b8f8deedac7b/containers/hbadeals/rsem2hbadeals.R#L234

Reply:
I think warnings 1 and 2 are from the function getvar() so that is less alarming, but warning 3 should not occur - looks like something is going on with some of the cores

Reply:
I am re running the job with the iterations you have been running and we can see how it behaves.

The job runs here: https://cloudos.lifebit.ai/public/jobs/5eb35fc31795470103dda4a3

Not merging until we check this.

Reply:
On a second thought, could the warning

```console
3: In mclapply(1:length(unique(countsData[, 1])), function(gene.number) { :
  scheduled cores 1, 5, 6, 8, 12, 28 did not deliver results, all values of the jobs will be affected
```
 have to do with any combination of the following:

- 10 observations (6 cases, 4 controls)
- 32 cores
- fewer iterations than usual

Reply:
Googling this warning lead me to a rather useful discussion here: https://github.com/pachterlab/sleuth/pull/94.

As `mclapply` is very memory-greedy, what I did in [`this test branch`](https://github.com/TheJacksonLaboratory/coronavirus-nf/tree/test-mcapply-fewer-cores) to  try to address the issue we were having was to:
- Allocate to the `hbadeals` Nextflow process 32 vCPUs (which come with 244.GB of memory) so that enough memory is available
https://github.com/TheJacksonLaboratory/coronavirus-nf/blob/b9a87bae04266835496ce10e4f5554eb35421e6b/conf/base.config#L71-L75
- Only allocate 4 vCPUs to `hbadeals` to restrict access to the whole machine's resources (_note: before that I was allocating 30 of 32 cpus to `hbadeals()`_)
https://github.com/TheJacksonLaboratory/coronavirus-nf/blob/5b84b780431b759f218859c1bbaedfbeaa42cbb3/main.nf#L1679-L1691

This has worked, here is the link to the job: 

I've captured `stderr` _(which in R, also warnings are redirected there \*sigh*)_
https://github.com/TheJacksonLaboratory/coronavirus-nf/blob/5b84b780431b759f218859c1bbaedfbeaa42cbb3/main.nf#L1679-L1691

After this I am getting only the first 2 warnings.
![image](https://user-images.githubusercontent.com/38183826/81275588-cd52f600-9049-11ea-9020-35e11c04e125.png)

I am testing now with more than 4 cpus allocated to not be wasteful of a 32 vCPUs machine and will commit this change in this branch, then I think it should be good for merging.

You can check the `stderr` log and the input data for `hbadeals()` in **results > hbadeals > cohort_1 .. > **
https://cloudos.lifebit.ai/public/jobs/5eb3bc3d1795470103deadba